src/Main.elm

```diff
@@ -1805,12 +1805,19 @@ encodeAttributeOptional encodeValue attributeOptional =
             encodeValue value
 
 
-decodeAttributeOptional : Json.Decode.Decoder a -> Json.Decode.Decoder (AttributeOptional a)
-decodeAttributeOptional decodeValue =
+decodeAttributeOptional : Maybe a -> Json.Decode.Decoder a -> Json.Decode.Decoder (AttributeOptional a)
+decodeAttributeOptional maybeNotNeeded decodeValue =
     Json.Decode.oneOf
         [ Json.Decode.null (AttributeNotNeeded Nothing)
         , decodeValue
-            |> Json.Decode.map (\a -> AttributeGiven a)
+            |> Json.Decode.map
+                (\a ->
+                    if Just a == maybeNotNeeded then
+                        AttributeNotNeeded Nothing
+
+                    else
+                        AttributeGiven a
+                )
```

Comment:
main change: `decodeAttributeOptional` is given a `Maybe a` to indicate what is a blank value. If the decoded value is such a blank value, then we should decode as `AttributeNotNeeded` instead of `AttributeGiven`

this helps decode old data with `"presence": { description: "" }` as `description = AttributeNotNeeded Nothing` (checkbox not ticked) instead of `description = AttributeGiven ""`  (checkbox ticked but value is invalid) https://github.com/choonkeat/tiny-form-fields/pull/9